### ISSUE TITLE
Update qty in admin order changes price to the wrong price from default store

### ISSUE THREAD

**Issue by digikeijs** [magento/magento2#15846]
**Summary of the issue**

When I use the option Update qty in the magento admin within an order the system changes the price to the default store price which is different then the store I am creating the order for. 

### Preconditions
magento 2.2.4, Linux, non sample data

### Steps to reproduce

1. Setup multistore
2. Default store price X, second store price Z. 
3. Then through the admin create a new order for the second store
4. Add a product with qty 5, update the qty to 8 and you will see the price is changed to the price of the default store (X)

### Expected result
Price stays the same when you update qty.

### Actual result
Price is changed to the default store price which is not correct

<!---
    Thank you for contributing to Magento.
    To help us process this issue we recommend that you add the following information:
     - Summary of the issue,
     - Information on your environment,
     - Steps to reproduce,
     - Expected and actual results,

![1](https://user-images.githubusercontent.com/34745091/40973951-59290686-68c6-11e8-8525-7e6ec3cf5939.JPG)
![2](https://user-images.githubusercontent.com/34745091/40973954-5ba4d2dc-68c6-11e8-81bc-e6084ed2045b.JPG)
![3](https://user-images.githubusercontent.com/34745091/40973956-5daca08c-68c6-11e8-9e0b-b2fb38f28381.JPG)

**Summary of the issue**

When I use the option Update qty in the magento admin within an order the system changes the price to the default store price which is different then the store I am creating the order for. 

**Information on your environment,**
magento 2.2.4, Linux, non sample data

**Steps to reproduce**

- Setup multistore, default store price X, second store price Z. Then through the admin create a new order for the second store, add a product with qty 5, update the qty to 8 and you will see the price is changed to the price of the default store (X)

**Expected and actual results**
Expected result is that the price stays the same when you update qty.
Actual result, the price is changed to the default store price which is not correct

<!---
    Thank you for contributing to Magento.
    To help us process this issue we recommend that you add the following information:
     - Summary of the issue,
     - Information on your environment,
     - Steps to reproduce,
     - Expected and actual results,

![1](https://user-images.githubusercontent.com/34745091/40973951-59290686-68c6-11e8-8525-7e6ec3cf5939.JPG)
![2](https://user-images.githubusercontent.com/34745091/40973954-5ba4d2dc-68c6-11e8-81bc-e6084ed2045b.JPG)
![3](https://user-images.githubusercontent.com/34745091/40973956-5daca08c-68c6-11e8-9e0b-b2fb38f28381.JPG)

**Comment by digikeijs** [comment]
@
![2](https://user-images.githubusercontent.com/34745091/40974003-810d0d32-68c6-11e8-8902-854330b1d230.JPG)
![1](https://user-images.githubusercontent.com/34745091/40974005-81383ffc-68c6-11e8-87cf-27a1069b5a1a.JPG)
![3](https://user-images.githubusercontent.com/34745091/40974006-8161f52c-68c6-11e8-9e73-bee46f36b0af.JPG)

**Comment by digikeijs** [comment]
@magento-engcom-team give me 2.2.4 instance

**Comment by magento-engcom-team** [comment]
Hi @digikeijs. Thank you for your request. I'm working on Magento 2.2.4 instance for you

**Comment by magento-engcom-team** [comment]
Hi @digikeijs, here is your [Magento instance](https://i-15846-2-2-4.engcom.dev.magento.com).
Admin access: [https://i-15846-2-2-4.engcom.dev.magento.com/admin](https://i-15846-2-2-4.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-backlog-nickolas** [comment]
@digikeijs, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by dnsv** [comment]
Duplicate of #14869.

**Comment by callumstar** [comment]
Hi @digikeijs, @engcom-backlog-nickolas and @magento-engcom-team,

Has this ever been resolved? Still getting this issue in 2.2.5? Has anybody found a workaround for this?

**Comment by SvanZui** [comment]
No waiting on the magento team to fix it. We have a workaround by creating a customer group with some discount rules and when we change groups it will be ok again. weird

**Comment by SvanZui** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @SAPontwerp. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @SAPontwerp, here is your [Magento instance](https://i-15846-2-2-6.engcom.dev.magento.com).
Admin access: [https://i-15846-2-2-6.engcom.dev.magento.com/admin](https://i-15846-2-2-6.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ihor-sviziev** [comment]
Hi @digikeijs,

This is duplicate of #14869

And not this issue was fixed both in 2.2-develop and 2.3-develop branches. It was even included into 2.2.6 release.

**Comment by ihor-sviziev** [comment]
Duplicate of #14869

**Comment by digikeijs** [comment]
@magento-engcom-team give me 2.2.4 instance

**Comment by magento-engcom-team** [comment]
Hi @digikeijs. Thank you for your request. I'm working on Magento 2.2.4 instance for you

**Comment by digikeijs** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @digikeijs. Thank you for your request. I'm working on Magento 2.2.6 instance for you
